lib/symlink.go

```diff
@@ -135,7 +135,23 @@ func ChangeProductSymlink(product Product, binVersionPath string, userBinPath st
 		err = CreateSymlink(binVersionPath, location)
 		if err == nil {
 			logger.Debugf("Symlink created at %q", location)
-			logger.Warnf("Run `export PATH=\"$PATH:%s\"` to append %q to $PATH", dirPath, location)
+
+			var isDirInPath bool = false
+			checkDirPath := strings.TrimRight(dirPath, "/")
+
+			for _, path := range strings.Split(os.Getenv("PATH"), ":") {
+				checkPath := strings.TrimRight(strings.Replace(path, "~", homedir, 1), "/")
```

Comment:
From what I can see/read, tilde expansion would only work during setting environment variables, meaning the resulting environment variable would contain the full path.
If an envionment variable does contain a tilde, then it won't be expanded and won't work as expected. I tested this a little locally and couldn't get it to expand.
So not sure if this replacement is required.
```suggestion
				checkPath := strings.TrimRight(path, "/")
```

Reply:
This is what I've got to test tilde in `PATH`:
```shell
> export PATH="~/.local/bin:~/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/usr/local/go/bin:/home/linuxbrew/.linuxbrew/bin:"

> echo $PATH
~/.local/bin:~/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/usr/local/go/bin:/home/linuxbrew/.linuxbrew/bin:

> hash -r

> which -a kitty

> type -a kitty
kitty is /home/georgii/.local/bin/kitty

> kitten --version
kitten 0.39.1 created by Kovid Goyal
```
So it seems to be expanded by Bash correctly.

Reply:
> `which` expectedly fails, because it's strictly POSIX-compliant and «It does not canonicalize path names»

Aha, this is why I went wrong :D 
Fair enough!